src/open-uri.c

```diff
@@ -585,6 +585,23 @@ app_exists (const char *app_id)
   return (info != NULL);
 }
 
+static char*
+build_host_path_from_doc_id(const char* doc_id, const char* doc_path)
+{
+  gchar *result;
+  g_autofree char *host_path = xdp_get_real_path_for_doc_id (doc_id);
+  g_autofree char *doc_path_regex = g_strconcat("/run/user/[0-9]+/doc/", doc_id, "/[^/]+(.*)", NULL);
```

Comment:
I think this could be avoided if `xdp_looks_like_document_portal_path` would add another out param which returns the path relative to the document.